go/parquet/internal/encoding/types.go

```diff
@@ -339,6 +339,12 @@ func (b *BufferWriter) Finish() *memory.Buffer {
 	return buf
 }
 
+// Release the underlying buffer and not allocate anything else. To re-use this buffer, Reset() or Finish() should be called
+func (b *BufferWriter) Release() {
```

Comment:
This had to be added, otherwise there's no way of completely "releasing the memory" without "allocating" a new buffer. (memory management related quotes used ironically)